src/app/zap-templates/templates/app/enums.zapt

```diff
@@ -2,19 +2,19 @@
 
 // Prevent multiple inclusion
 #pragma once
-
+#include <stdint.h>
 // ZCL enums
 
 {{#zcl_enums}}
 {{#unless (isStrEqual label "Status")}}
 {{#unless (isStrEqual label "ReportingDirection")}}
 
 // Enum for {{label}}
-typedef enum {
+enum EmberAf{{asType label}}:uint32_t{
```

Comment:
I suspect we want to change ZAP to allow querying either the range of enum values or the type to hold he enum directly.  @vivien-apple thoughts?

Reply:
This will require us to modify the [zap engine](https://github.com/project-chip/zap) itself and is beyond the scope of this PR. Adding `uint32_t` suffices for fixing the wrong overload function call in the sdk.

Reply:
It suffices, but it also (potentially?) increases the sizes of any data structures storing those enums, no?  At the very least we need something tracking the followup to look into whether we should have better size bounds here.

Reply:
https://github.com/project-chip/connectedhomeip/issues/7394